src/pages/workspaces/WorkspacesList/WorkspaceUserActions.ts

```diff
@@ -0,0 +1,24 @@
+import { createContext } from 'react';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+
+export const WorkspaceUserActionsContext = createContext<WorkspaceUserActionsState>({
+  userActions: {
+    creatingNewWorkspace: false,
+  },
+  setUserActions: () => {},
+});
+
+export interface WorkspaceUserActionsState {
+  userActions: WorkspaceUserActions;
+  setUserActions: (actions: Partial<WorkspaceUserActions>) => void;
+}
+
+export interface WorkspaceUserActions {
+  creatingNewWorkspace: boolean;
+  cloningWorkspaceId?: string;
+  deletingWorkspaceId?: string;
+  lockingWorkspaceId?: string;
+  leavingWorkspaceId?: string;
+  requestingAccessWorkspaceId?: string;
+  sharingWorkspace?: WorkspaceWrapper;
```

Comment:
Why do all other fields contain the workspace ID and this contains the workspace itself. Ideally, they would all be consistent.

Reply:
I didn't want to untangle the logic this time around for sake of getting the other changes done, so I left that part of the existing implementation in place.